### PR TITLE
MAINT: Expire deprecations

### PR DESCRIPTION
**Pull request by m-richards**
xref #2263. This definitely needs a changelog, haven't done that yet.

### CODE REVIEW

File: geopandas/tools/sjoin.py [hunk 1]
```diff
@@ -196,27 +175,21 @@ def _geom_predicate_query(left_df, right_df, predicate, distance):
         DataFrame with matching indices in
         columns named `_key_left` and `_key_right`.
     """
-    with warnings.catch_warnings():
-        # We don't need to show our own warning here
-        # TODO remove this once the deprecation has been enforced
-        warnings.filterwarnings(
-            "ignore", "Generated spatial index is empty", FutureWarning
```
Comment (jorisvandenbossche) [inline geopandas/tools/sjoin.py:203]: What was this warning about? (that's for a warning we already removed in the meantime?)

**Review by jorisvandenbossche — APPROVED** [review]
Looks good, thanks a lot!

File: geopandas/tools/sjoin.py (hunk 1 above)
Comment (m-richards) [reply geopandas/tools/sjoin.py:203]: Yep, https://github.com/geopandas/geopandas/pull/1444/ in geopandas/base.py::_generate_sindex

**Review by martinfleis — APPROVED** [review]
Thanks!

**Pull request merged**